src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Documents/EditorDocumentManagerBase.cs

```diff
@@ -197,22 +206,25 @@ public sealed override void RemoveDocument(EditorDocument document)
 
             ForegroundDispatcher.AssertForegroundThread();
 
-            var key = new DocumentKey(document.ProjectFilePath, document.DocumentFilePath);
-            if (_documentsByFilePath.TryGetValue(document.DocumentFilePath, out var documents))
+            lock (_lock)
```

Comment:
Good catch.

Maybe not for this PR, but would this be better if we did `ConcurrentDictionary` instead of `Dictionary`? I'm thinking it might not work because this call probably needs the TryGetValue and Remove calls on `_documentsByFilePath` to be atomic, but it'd be a shame to use locks if `ConcurrentDictionary` would do (to avoid mistakes like the one you're fixing here).

Reply:
Good idea! Yeah, I would assume those operations would have to be atomic, but if we could avoid using locks here somehow that would be ideal.